corva/application.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, List, Optional
+
+from corva.middleware.unpack_context import unpack_context
+from corva.models.base import BaseContext
+from corva.types import MIDDLEWARE_CALL_TYPE, MIDDLEWARE_TYPE
+
+
+def wrap_call_in_middleware(
+     call: Callable,
+     middleware: Optional[List[MIDDLEWARE_TYPE]] = None
+) -> MIDDLEWARE_CALL_TYPE:
+    def wrapper_factory(mw, call):
+        def wrapper(ctx):
+            return mw(ctx, call)
+
+        return wrapper
+
+    middleware = middleware or []
+
+    for mw in reversed(middleware):
+        call = wrapper_factory(mw, call)
+
+    return call
+
+
+class Corva:
+    def __init__(
+         self,
+         *,
```

Comment:
We're not doing anything with the * here? Can we do *args at least?

Reply:
wanted to make arguments to Corva class keyword only, so they dont get mixed, when there are a lot of settings

Reply:
That is generating more questions than helps a user. Can we go with just one keyword parameter we have for now?

Reply:
fixed